compiler/rustc_hir_typeck/src/expectation.rs

```diff
@@ -77,39 +81,31 @@ impl<'a, 'tcx> Expectation<'tcx> {
         }
     }
 
-    /// Resolves `expected` by a single level if it is a variable. If
-    /// there is no expected type or resolution is not possible (e.g.,
-    /// no constraints yet present), just returns `self`.
-    fn resolve(self, fcx: &FnCtxt<'a, 'tcx>) -> Expectation<'tcx> {
+    pub(super) fn structurally_resolve(
```

Comment:
I decided to rename these methods to make it clear that they structurally resolve.